test-src/org/pentaho/metastore/test/MetaStoreFactoryTest.java

```diff
@@ -296,6 +297,29 @@ private Cube generateCube() {
     return cube;
   }
 
+  public void testDisallowEmptyName() throws Exception {
+    IMetaStore metaStore = new MemoryMetaStore();
+    MetaStoreFactory<MyElement> factory = new MetaStoreFactory<MyElement>( MyElement.class, metaStore, "custom" );
+
+    try {
+      MyElement element = new MyElement();
+      element.setName( null );
+      factory.saveElement( element );
+      fail( "Saved illegal element (name == null)" );
+    } catch ( MetaStoreException e ) {
+      assertNotNull( e );
+    }
+
+    try {
+      MyElement element = new MyElement();
+      element.setName( "" );
```

Comment:
What about the test case where the element name is "   " (to validate the name is being trimmed). We may also need a test case checking for duplicates where the names are the same except the whitespace...